addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/NestAuthorizer.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nest.internal;
+
+import java.io.IOException;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.nest.NestBindingConstants;
+import org.openhab.binding.nest.internal.config.NestBridgeConfiguration;
+import org.openhab.binding.nest.internal.data.AccessTokenData;
+import org.openhab.binding.nest.internal.exceptions.InvalidAccessTokenException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * Retrieves the Nest access token using the OAuth 2.0 protocol using pin-based authorization.
+ *
+ * @author David Bennett - Initial contribution
```

Comment:
I don't think I wrote this since it is an entirely new file :)

Reply:
It used to be named NestAccessToken. Back then it also accessed the file. But after refactoring the only function that remained was getting the token. So I renamed it because it was confusing with the accessToken stored in the account config. So it does share some history with you. ;-)